metricflow/engine/metricflow_engine.py

```diff
@@ -523,13 +544,28 @@ def get_dimension_values(  # noqa: D
                 group_by_names=[get_group_by_values],
                 time_constraint_start=time_constraint_start,
                 time_constraint_end=time_constraint_end,
+                query_type=MetricFlowQueryType.DIMENSION_VALUES,
             )
         )
         result_dataframe = query_result.result_df
         if result_dataframe is None:
             return []
+        return [str(val) for val in result_dataframe[get_group_by_values]]
 
-        # Process the dimension values
-        result_dataframe.dropna(inplace=True)
-        dimension_values = [str(val) for val in result_dataframe[get_group_by_values]]
-        return dimension_values
+    @log_call(module_name=__name__, telemetry_reporter=_telemetry_reporter)
+    def explain_get_dimension_values(  # noqa: D
```

Comment:
there has to be a better name here, but I can't figure it out. Any advice is appreciated or we can leave it as it 🙈